src/System.Security.Cryptography.Pkcs/tests/EnvelopedCms/KeyAgreeRecipientInfoTests.cs

```diff
@@ -297,6 +284,45 @@ public static void TestKeyAgreeOtherKeyAttribute_FixedValue_Ski()
             Assert.Equal<byte>(expectedAsnData, asnData.RawData);
         }
 
+        [Fact]
+        public static void TestMultipleKeyAgree_ShortNotation()
+        {
+            // KeyAgreement recipients are defined in RFC 2630 as
+            //
+            // KeyAgreeRecipientInfo::= SEQUENCE {
+            //      version CMSVersion,  --always set to 3
+            //      originator[0] EXPLICIT OriginatorIdentifierOrKey,
+            //      ukm[1] EXPLICIT UserKeyingMaterial OPTIONAL,
+            //      keyEncryptionAlgorithm KeyEncryptionAlgorithmIdentifier,
+            //      recipientEncryptedKeys RecipientEncryptedKeys }
+            //
+            // RecipientEncryptedKeys::= SEQUENCE OF RecipientEncryptedKey
+            //
+            // RecipientEncryptedKey::= SEQUENCE {
+            //      rid KeyAgreeRecipientIdentifier,
+            //      encryptedKey EncryptedKey }
+            //
+            // When RecipientEncryptedKeys has more than one sequence in it, then different KeyAgreement recipients are created, where each
+            // recipient hold the information from one RecipientEncryptedKey  
+
```

Comment:
I'm assuming that the important, but unstated, part is that this encoded message has one KeyAgreeRecipientInfo which has two EncryptedKeys values?